### ISSUE TITLE
Support for criteria that checkouts the version of the code that triggered the change

### ISSUE THREAD

**Issue by orviz** [EOSC-synergy/sqaaas-api-server#32]
In principle, this can be tackled through the current API spec, with no additional property or change. If `project_repos:repo:branch` is empty, then the API should assume that this is the case.

The correct format to be added in the config.yml is the one that avoids the definition of the `repos:<repo_name>`, which is [supported by JePL](https://github.com/indigo-dc/jenkins-pipeline-library/issues/142).

**Comment by orviz** [comment]
For the time being, the API will rely on the format:
```yaml
sqa_criteria:
  <criterion>:
    repos:
      this_repo:
        commands: ..
```

as it is best supported by JePL. Note that `this_repo` is the agreed placeholder that represents the repo where the pipeline is stored.

**Comment by orviz** [comment]
Released in 1.1.1
